enhancements/operator-dependency-resolution.md

```diff
@@ -0,0 +1,205 @@
+
+That being said, these additional types will need to be implemented as they become more clear. For now, `dependencies.yaml` will not include any explicit reference to provided APIs.
+
+### Cluster Admin UX
+
+Given the changes to the way OLM will resolve dependencies, it is also necessary to push more detailed data to cluster administrators about what dependencies were installed (or will be installed) and why. To that end, we will continue to use the InstallPlan status to push detailed resolution information to the user. This will include data about what dependencies were installed along with information about where child dependencies came from (essentially, the result of the resolver in detail).
+
+There was also some initial discussion around what the user workflow would be to edit/override dependencies. Given the current state of the APIs that exist (specifically the Subscription and InstallPlan APIs), those would require extensive changes to create a trivial override for dependencies. Even then, there is a configurability problem where a single install overriding a dependency may not be enough and a cluster admin may want to create a cluster wide or namespaced override that is persistent and editable. Initially, there was some discussion about the possibility of a dependency Override API or the possibility of using the Operator API for this purpose, but those discussions are ultimately outside the scope of the immediate (3+ month) timeframe and scope of this enhancement proposal, and will be taken up further in the future.
+
+### User Stories
+
+#### Explicit Operator Dependency Metadata
+
+As an operator author, I want to specify dependencies explicitly in the bundle. For example, I would like to specify that my operator depends on etcd operator version 0.9.0.
+* Explicit operator dependency is natural, intuitive and safe to understand
+* Avoid unexpected situation where two different operators owning the same CRD/API
+
+Acceptance Criteria:
+I can specify explicit dependencies in a file in the bundle and it can be included with other manifests and metadata in the bundle image.
+
+Internal details:
+Currently, the plan is to have `dependencies.yaml` as a centralized location for explicit dependency specification. However, we may plan to include that information in Dockerfile/manifest list so that information can be query without unpack the bundle image to see its content.
+
+#### Explicit Operator Dependency Validation
+
+As an operator author, I want to validate dependencies solution in the bundle. For example, I would like to make sure OLM will resolve and install all the dependencies I specify in the bundle correctly.
+* Allow operator authors to test dependency resolution for their operators before publishing
+
+Acceptance Criteria:
+I can use some toolings to list the operators that will be installed for my operator that will meet all of my dependency requirements that I specify in the bundle.
```

Comment:
It would be also very nice to actually do that on cluster with all catalogs registered. An equivalent to `yum --dry-run`.

Reply:
I added a note in the Internal details to select this option.